test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/SqlProjects/SqlProjectsServiceTests.cs

```diff
@@ -3,91 +3,179 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 //
 
-using System;
 using System.IO;
 using System.Threading.Tasks;
 using Microsoft.SqlServer.Dac.Projects;
-using Microsoft.SqlTools.Hosting.Protocol;
 using Microsoft.SqlTools.ServiceLayer.IntegrationTests.Utility;
 using Microsoft.SqlTools.ServiceLayer.SqlProjects;
 using Microsoft.SqlTools.ServiceLayer.SqlProjects.Contracts;
-using Moq;
+using Microsoft.SqlTools.ServiceLayer.Test.Common.RequestContextMocking;
 using NUnit.Framework;
 
 namespace Microsoft.SqlTools.ServiceLayer.IntegrationTests.SqlProjects
 {
     public class SqlProjectsServiceTests
     {
         [Test]
-        public async Task TestOpenCloseProject()
+        public async Task TestErrorDuringExecution()
         {
-            // setup
+            SqlProjectsService service = new();
+            string projectUri = await service.CreateSqlProject(); // validates result.Success == true
+
+            // Validate that result indicates failure when there's an exception
+            MockRequest<SqlProjectResult> requestMock = new();
+            await service.HandleNewSqlProjectRequest(new NewSqlProjectParams()
+            {
+                ProjectUri = projectUri,
+                SqlProjectType = ProjectType.SdkStyle
+
+            }, requestMock.Object);
 
-            var newSdkMock = new MockRequest<SqlProjectResult>();
-            var newLegacyMock = new MockRequest<SqlProjectResult>();
-            var closeMock = new MockRequest<SqlProjectResult>();
-            var openMock = new MockRequest<SqlProjectResult>();
+            Assert.IsFalse(requestMock.Result.Success);
+            Assert.IsTrue(requestMock.Result.ErrorMessage!.Contains("Cannot create a new SQL project"));
+        }
 
+        [Test]
+        public async Task TestOpenCloseProject()
+        {
+            // Setup
             string sdkProjectUri = TestContextHelpers.GetTestProjectPath(nameof(TestOpenCloseProject) + "Sdk");
             string legacyProjectUri = TestContextHelpers.GetTestProjectPath(nameof(TestOpenCloseProject) + "Legacy");
```

Comment:
do the projects created by these tests get cleaned up after the test runs?

Reply:
Not currently.  Do you know of any infra that already exists in tools service that can be used for automatically cleaning folders up?  I'm thinking along the lines of a folder that's used for each test run that can just be nuked as a cleanup step.

Reply:
not that I know of :( Schema compare and dacpac tests handle their own test file cleanup.

Reply:
I signed off on all the other changes and so ok if you want handle the cleanup for test files in a separate PR later